spec/unit/builders/ProjectBuilder.spec.js

```diff
@@ -182,34 +181,34 @@ describe('ProjectBuilder', () => {
 
             builder.build({});
 
-            expect(spawnSpy).toHaveBeenCalledWith(path.join(rootDir, 'gradlew'), testArgs, jasmine.anything());
+            expect(execaSpy).toHaveBeenCalledWith(path.join(rootDir, 'gradlew'), testArgs, jasmine.anything());
         });
 
         it('should reject if the spawn fails', () => {
-            const errorMessage = 'ERROR: Failed to spawn';
-            spawnSpy.and.returnValue(Q.reject(errorMessage));
+            const errorMessage = 'Test error';
+            execaSpy.and.rejectWith(new Error(errorMessage));
 
             return builder.build({}).then(
                 () => fail('Unexpectedly resolved'),
-                err => {
-                    expect(err).toBe(errorMessage);
+                error => {
+                    expect(error.message).toBe(errorMessage);
                 }
             );
         });
 
         it('should check the Android target if failed to find target', () => {
             const checkReqsSpy = jasmine.createSpyObj('check_reqs', ['check_android_target']);
-            const errorMessage = 'ERROR: failed to find target with hash string';
+            const testError = 'failed to find target with hash string';
 
             ProjectBuilder.__set__('check_reqs', checkReqsSpy);
-            checkReqsSpy.check_android_target.and.returnValue(Q.resolve());
-            spawnSpy.and.returnValue(Q.reject(errorMessage));
```

Comment:
The `Q` changes I think should done in the dropping Q PR : https://github.com/apache/cordova-android/pull/833, but this type of work kinda blends together.

Reply:
The old code used `superspawn` which returns `Q` Promises, so it was natural for the test stubs to return `Q` Promises too. Now that we stub `execa`, we should not return `Q` Promises from the stub anymore, as the interface provided by them is not a subset of the type returned from `execa`. That could lead to code passing the tests but failing in reality.